packages/payment-detection/test/erc20/transferable-receivable.test.ts

```diff
@@ -0,0 +1,317 @@
+          timestamp: 1,
+        },
+        // Correct reference and contract
+        {
+          contractAddress: erc20LocalhostContractAddress,
+          to: paymentAddress,
+          from: '0x186e7fe6c34ea0eca7f9c2fd29651fc0443e3f29',
+          amount: '400',
+          feeAmount: '0',
+          reference: hashedReference,
+          block: 9610470,
+          txHash: '0x2f7b4752aa259166c038cd9073056c5979760cf0eea55d093fca2095c229313b',
+          feeAddress: feeAddress,
+          gasPrice: '',
+          gasUsed: '',
+          timestamp: 1,
+        },
+        // Correct reference and contract with different receivable owner
+        {
+          contractAddress: erc20LocalhostContractAddress,
+          to: '0x186e7fe6c34ea0eca7f9c2fd29651fc0443e3f30',
+          from: '0x186e7fe6c34ea0eca7f9c2fd29651fc0443e3f29',
+          amount: '600',
+          feeAmount: '0',
+          reference: hashedReference,
+          block: 9610471,
+          txHash: '0x2f7b4752aa259166c038cd9073056c5979760cf0eea55d093fca2095c229313b',
+          feeAddress: feeAddress,
+          gasPrice: '',
+          gasUsed: '',
+          timestamp: 1,
+        },
+      ],
+    };
+
+    const clientMock = {
+      GetPaymentsAndEscrowState: jest.fn().mockImplementation(({}) => ({
+        payments: [],
+        escrowEvents: [],
+      })),
+      GetPaymentsAndEscrowStateForReceivables: jest.fn().mockImplementation(({ reference }) => ({
+        payments: paymentsMockData[reference] || [],
+        escrowEvents: [],
+      })),
+      GetLastSyncedBlock: jest.fn(),
+      GetSyncedBlock: jest.fn(),
+    };
+
+    erc20TransferableReceivable = new ERC20TransferableReceivablePaymentDetector({
+      advancedLogic: mockAdvancedLogic,
+      currencyManager,
+      getSubgraphClient: jest.fn().mockImplementation(({}) => {
+        return clientMock;
+      }),
+    });
+
+    const graphRetriever = new TheGraphInfoRetriever(clientMock, CurrencyManager.getDefault());
+    const allNetworkEvents = await graphRetriever.getReceivableEvents({
+      paymentReference: shortReference,
+      contractAddress: erc20LocalhostContractAddress,
+      toAddress: '',
+      eventName: PaymentTypes.EVENTS_NAMES.PAYMENT,
+      paymentChain: paymentData.network,
+    });
+
+    const transferEvents = allNetworkEvents.paymentEvents;
+    expect(transferEvents).toHaveLength(2);
+    expect(transferEvents[0].amount).toEqual('400');
+    expect(transferEvents[0].name).toEqual('payment');
+    expect(transferEvents[0].parameters?.to).toEqual(utils.getAddress(paymentData.to));
+    expect(transferEvents[0].parameters?.block).toEqual(9610470);
+
+    expect(transferEvents[1].amount).toEqual('600');
+    expect(transferEvents[1].name).toEqual('payment');
+    expect(transferEvents[1].parameters?.to).toEqual(
+      utils.getAddress('0x186e7fe6c34ea0eca7f9c2fd29651fc0443e3f30'),
+    );
+    expect(transferEvents[1].parameters?.block).toEqual(9610471);
+  });
+});
```

Comment:
Please add a few more tests (see https://github.com/00labs/requestNetwork/blob/pr/mliu/1033/packages/payment-detection/test/erc20/fee-proxy-contract.test.ts for inspiration)

* can getBalance using proxy info retreiver
* can getBalance using thegraph info retreiver
* cannot createExtensionsDataForDeclareSentPayment (just to show that the declarative functions are not implemented)